packages/aws-cdk-lib/aws-ec2/lib/instance.ts

```diff
@@ -697,6 +706,14 @@ export class Instance extends Resource implements IInstance {
     this.role.addToPrincipalPolicy(statement);
   }
 
+  /**
+   * Adds a network interface to the instance.
+   */
+  public addNetworkInterface(networkInterface: INetworkInterface, options: AddNetworkInterfaceOptions = {}) {
```

Comment:
can a same network interface be attached to two instance? i think not.. let me know if that's not the case.
instead of taking a network interface as an argument, create a new one with and take in the parameters that are required to create n/w interface